esm_runscripts/compute.py

```diff
@@ -532,14 +538,33 @@ def copy_tools_to_thisrun(config):
                 fromdir, scriptsdir, tfile, gconfig, "additional file"
             )
 
-        restart_command = (
-            "cd "
-            + scriptsdir
-            + "; "
-            + "esm_runscripts "
-            + gconfig["original_command"].replace("-U", "")
-            + " --no-motd "
-        )
+        # remove the update option otherwise it will enter an infinite loop
+        original_command = gconfig["original_command"]
+        options_to_remove = [" -U ", " --update "]
+        for option in options_to_remove:
+            original_command = original_command.replace(option, " ")
+        
+        # Before resubmitting the esm_runscripts, the path of the runscript
+        # needs to be modified. Remove the absolute/relative path
+        runscript_absdir, runscript = os.path.split(gconfig['runscript_abspath'])
+        original_command_list = original_command.split()
+        new_command_list = []
+        for command in original_command_list:
+            # current command will contain the full path, so replace it with
+            # the YAML file only since we are going to execute it from the 
+            # `scriptsdir` now
+            if runscript in command:
+                # gconfig['scriptname'] or `runscript` only contains the YAML file name
+                command = runscript 
+            new_command_list.append(command)
```

Comment:
I think I am not understanding why we need an absolute path for the runscript here. Would this not be running always from the `<exp_path/scrips>` directory?

Reply:
Hi @mandresm,

### 1) Explanation:
Unfortunately there are some hard-coded stuff in `esm_runscripts`. Take this:

in `cli.py`: 
```python
command_line_config["started_from"] = os.getcwd()
```

This is later on used in `compute.py`:
```python
# Directory where the original runscript is
fromdir = os.path.realpath(gconfig["started_from"])
# Directory where the runscript is copied to
scriptsdir = os.path.realpath(gconfig["experiment_scripts_dir"])
```
and there are checks
```python
if (fromdir == scriptsdir) and not gconfig["update"]:
```
Sure, the the `esm_runscripts` will enter the experiment directory and run the script in `<exp_path/scrips>`, in the default one this does not happen. This is why I added `gconfig['runscript_abspath']`, so that I can carry the information of where the original runscript resides.

Indeed, the cool thing is in the `restart_command`, 
`restart_command = f"cd {scriptsdir}; esm_runscripts {new_command}"`
in now reduntant. `restart_command = f"esm_runscripts {scriptsdir}/{new_command}"` should also work. So, the idea is there is no need to change the directory to the where script lies, to run it. 

### 2) Bugfix or feature:
This is a good question. To me it looks more like a bugfix since `--update` is now fixed. Calling the scripts from another directory could be more like a feature though.

Reply:
Thanks for the explanation :)